float32 readyToExchange
float32 readyToExchange2
